publisher_scaled_joint_trajectory_controller:
  ros__parameters:

    controller_name: "scaled_joint_trajectory_controller"

    goal_names: ["idle_pose", "measurement_pose"]
    idle_pose:
      positions: [0.0, -1.57, -1.57, -1.57, 1.57, 0.0]
    measurement_pose:
      positions: [0.0, -0.872664626, -2.2689280276, -1.57, 1.57, 0.0]
    
    # This sets the time in which a move should be executed if only position is specified
    # Format: [seconds, nanoseconds]
    movement_duration: [4, 0]
    
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint

    check_starting_point: true
    starting_point_limits:
      shoulder_pan_joint: [-0.2,0.2]
      shoulder_lift_joint: [-1.7,-1.4]
      elbow_joint: [-1.7,-1.4]
      wrist_1_joint: [-1.7,-1.4]
      wrist_2_joint: [1.4,1.7]
      wrist_3_joint: [-0.2,0.2]